### PR TITLE
Moredelays

### PR DESCRIPTION
**Pull request by baggepinnen**
Some fixes and generalizations for delay systems

### CODE REVIEW

File: src/connections.jl [hunk 1]
```diff
@@ -135,7 +135,10 @@ function Base._cat_t(::Val{2}, T::Type{<:LTISystem}, X...)
 end
 
 # Used in typed_hvcat
-Base.typed_hcat(::Type{T}, X...) where {T<:LTISystem} = hcat(convert.(T, X)...)
+function Base.typed_hcat(::Type{T}, X...) where {T<:LTISystem}
+    S = promote_type(typeof.(X)...)
+    hcat(convert.(S, X)...)
+end
```
Comment (mfalt) [inline src/connections.jl]: I'm not sure this is actually what we want to do.

File: src/types/DelayLtiSystem.jl [hunk 2]
```diff
@@ -156,10 +156,14 @@ function feedback(sys1::DelayLtiSystem, sys2::DelayLtiSystem)
...
 end
 
-function delay(tau::S, T::Type{<:Number}=Float64) where S
+function delay(T::Type{<:Number}, tau)
     return DelayLtiSystem(ControlSystems.ss([zero(T) one(T); one(T) zero(T)]), [T(tau)])
 end
 
+function delay(tau::S) where S
+    return DelayLtiSystem(ControlSystems.ss([zero(S) one(S); one(S) zero(S)]), [S(tau)])
+end
```
Comment (mfalt) [inline src/types/DelayLtiSystem.jl]: I agree with the argument order above, but I'm nbot sure we should drop the Float default. For example, the lsim above will break for a `delay(1)` right?

File: src/connections.jl (hunk 1 above)
Comment (baggepinnen) [reply src/connections.jl]: No, probably me who messed up while experimenting

File: src/types/DelayLtiSystem.jl (hunk 2 above)
Comment (baggepinnen) [reply src/types/DelayLtiSystem.jl]: I defaulted to `promote_type(Float64, typeof(tau))` instead.

**Comment by baggepinnen** [comment]
Ready to pull and tag

File: src/connections.jl (hunk 1 above)
Comment (baggepinnen) [reply src/connections.jl]: See here?

**Pull request merged**